music21/metadata/__init__.py

```diff
@@ -96,161 +184,499 @@ class Metadata(base.Music21Object):
+        return self._get(key, isCustom=True)
+
+    def addCustom(self, key: str, value: t.Union[t.Any, t.Iterable[t.Any]]):
+        self._add(key, value, isCustom=True)
+
+    def setCustom(self, key: str, value: t.Union[t.Any, t.Iterable[t.Any]]):
+        self._set(key, value, isCustom=True)
+
+    def getAllNamedValues(self, skipContributors=False) -> t.List[t.Tuple[str, ValueType]]:
+        '''
+        Returns all values stored in this metadata as a list of (nsKey, value) tuples.
+        nsKeys with multiple values will appear multiple times in the list (rather
+        than appearing once, with a value that is a list of values).
+        The tuple's first element is either of the form 'namespace:name', or a
+        custom key (with no form at all).
+
+        >>> md = metadata.Metadata()
+        >>> md.add('composer', 'Jeff Bowen')
+        >>> md.add('librettist', 'Hunter Bell')
+        >>> md.add('title', 'Other World')
+        >>> md.addCustom('excerpt-start-measure', 1234)
+        >>> all = md.getAllNamedValues()
+        >>> all
+        [('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Jeff Bowen>),
+        ('marcrel:LBT', <music21.metadata.primitives.Contributor librettist:Hunter Bell>),
+        ('dcterms:title', <music21.metadata.primitives.Text Other World>),
+        ('excerpt-start-measure', <music21.metadata.primitives.Text 1234>)]
+        >>> allNonContributors = md.getAllNamedValues(skipContributors=True)
+        >>> allNonContributors
+        [('dcterms:title', <music21.metadata.primitives.Text Other World>),
+        ('excerpt-start-measure', <music21.metadata.primitives.Text 1234>)]
+        '''
+        allOut: t.List[t.Tuple[str, ValueType]] = []
+
+        valueList: t.List[ValueType]
+        for nsKey, valueList in self._metadata.items():
+            if skipContributors and self._isContributorNSKey(nsKey):
+                continue
+
+            value: ValueType
+            for value in valueList:
+                allOut.append((nsKey, value))
+
+        return allOut
+
+    def getAllContributorNamedValues(self) -> t.List[t.Tuple[str, ValueType]]:
+        '''
+        Returns all contributors stored in this metadata as a list of (nsKey, value) tuples.
+        The tuple's first element will be of the form 'namespace:name'.
+
+        >>> md = metadata.Metadata()
+        >>> md['title'] = ['Dimitrij', 'False Dmitry']
+        >>> md.composer = 'Antonín Dvořák'
+        >>> md.opusNumber = 64
+        >>> md.add('librettist', 'Marie Červinková-Riegrová')
+        >>> md['otherContributor'] = (metadata.Contributor(role='based on plot by',
+        ...                               name ='Ferdinand Mikovec'),
+        ...                           metadata.Contributor(role='original partial plot by',
+        ...                               name='Friedrich Schiller'))
+        >>> md.addCustom('composer', 'Not a contributor')
```

Comment:
`'composer'` is a particularly confusing example to show as a custom field for "Not a contributor" -- better to add something like `'averageDuration', '180 minutes'` and use the fact that it's a doctest to explain in prose that non-contributor fields will not appear.

Reply:
Done.